src/sagas/getDataSagas.ts

```diff
@@ -1,15 +1,20 @@
 import { put, call, takeEvery } from 'redux-saga/effects'
-import { fetchPostsFailure, fetchPostsSuccess } from '../features/posts/postsSlice'
-import { postsData } from "../redux/postsData";
+import { fetchPostsFailure, fetchPostsSuccess, IPost } from '../features/posts/postsSlice'
 
-const delay = (ms: number) => new Promise(res => {
-    setTimeout(res, ms)
-})
+type dataPost = {
+    count: number, 
+    next: null, 
+    previous: null, 
+    results: Array<IPost>
+}
 
 export function* fetchPosts() {
     try {
-        yield call(delay, 100)
-        yield put(fetchPostsSuccess(postsData))
+        const response: Response = yield call(() => new Promise(res => {
+            res(fetch(`https://studapi.teachmeskills.by/blog/posts/?limit=70&offset=0`)) 
+        }))
```

Comment:
Оборачивать в дополнительный промисс нет необходимости, лучше вынести саму функцию-запрос в отдельную функцию и обработать ошибки там
```suggestion
        const fetchPosts = () => {
          try {
           const response = fetch(`https://studapi.teachmeskills.by/blog/posts/?limit=70&offset=0`)) 
          }
           …
        }
        const data: Data = yield call(fetchPosts)
```